src/qt/optionsmodel.cpp

```diff
@@ -637,4 +617,49 @@ void OptionsModel::checkAndMigrate()
     if (settings.contains("addrSeparateProxyTor") && settings.value("addrSeparateProxyTor").toString().endsWith("%2")) {
         settings.setValue("addrSeparateProxyTor", GetDefaultProxyAddress());
     }
+
+    // Migrate and delete legacy GUI settings that have now moved to <datadir>/settings.json.
+    auto migrate_setting = [&](OptionID option, const QString& qt_name, const std::string& name) {
+        if (!settings.contains(qt_name)) return;
+        QVariant value = settings.value(qt_name);
+        if (node().getPersistentSetting(name).isNull()) {
+            if (option == ProxyIP) {
+                ProxySetting parsed = ParseProxyString(value.toString());
+                setOption(ProxyIP, parsed.ip);
+                setOption(ProxyPort, parsed.port);
+            } else if (option == ProxyIPTor) {
+                ProxySetting parsed = ParseProxyString(value.toString());
+                setOption(ProxyIPTor, parsed.ip);
+                setOption(ProxyPortTor, parsed.port);
+            } else {
+                setOption(option, value);
+            }
+        }
+        settings.remove(qt_name);
+    };
+
+    migrate_setting(DatabaseCache, "nDatabaseCache", "dbcache");
+    migrate_setting(ThreadsScriptVerif, "nThreadsScriptVerif", "par");
+#ifdef ENABLE_WALLET
+    migrate_setting(SpendZeroConfChange, "bSpendZeroConfChange", "spendzeroconfchange");
+    migrate_setting(ExternalSignerPath, "external_signer_path", "signer");
+#endif
+    migrate_setting(MapPortUPnP, "fUseUPnP", "upnp");
+    migrate_setting(MapPortNatpmp, "fUseNatpmp", "natpmp");
+    migrate_setting(Listen, "fListen", "listen");
+    migrate_setting(Server, "server", "server");
+    migrate_setting(PruneSize, "nPruneSize", "prune");
+    migrate_setting(Prune, "bPrune", "prune");
+    migrate_setting(ProxyIP, "addrProxy", "proxy");
+    migrate_setting(ProxyUse, "fUseProxy", "proxy");
```

Comment:
Alright, if this PR settles as it is (to store "use proxy" implicitly), then this is ok IMO. If it is decided to store "use proxy" explicitly then this will be changed anyway.

Can be marked as resolved.